build-settings-logic/src/main/kotlin/atomicfu-dependency-resolution-management.settings.gradle.kts

```
@@ -0,0 +1,87 @@
+import org.gradle.kotlin.dsl.maven
+
+pluginManagement {
+    repositories {
+        mavenCentral()
+        gradlePluginPortal()
+
+        val additionalRepositoryProperty = providers.gradleProperty("kotlin_repo_url")
+        if (additionalRepositoryProperty.isPresent) {
+            maven(url = uri(additionalRepositoryProperty.get()))
+            logger.info("A custom Kotlin repository ${additionalRepositoryProperty.get()} was added")
+        }
+
+        /*
+         * This property group is used to build kotlinx.atomicfu against Kotlin compiler snapshots.
+         * When build_snapshot_train is set to true, mavenLocal and Sonatype snapshots are added to a repository list
+         * (the former is required for AFU and public, the latter is required for compiler snapshots).
+         * DO NOT change the name of these properties without adapting the kotlinx.train build chain.
+         */
+        val buildSnapshotTrainGradleProperty = providers.gradleProperty("build_snapshot_train")
+        if (buildSnapshotTrainGradleProperty.isPresent) {
+            maven(url = uri("https://oss.sonatype.org/content/repositories/snapshots"))
+        }
```

It's usually better to keep the build config as stable as possible. This avoids surprises, and helps with Gradle caching (configuration cache in this case).

[There was an issue a couple of weeks ago caused by conditional Maven repos](https://jetbrains.slack.com/archives/C4U955N6B/p1721204053175179).

Could the Maven Central snapshots be added without the conditional logic? The content could be filtered to help performance, so that Gradle will only look for snapshots in it.

```suggestion
        maven("https://oss.sonatype.org/content/repositories/snapshots") {
            mavenContent { snapshotsOnly }
        }
```

---

@adam-enko I started exploring this idea and came up to an interesting conclusion, that `https://oss.sonatype.org/content/repositories/snapshots` is not as it was expected. Instead, in the train builds, we use only Maven local to publish intermediate artifacts.

---

Ah okay, so Maven Central Snapshots is not used at all? If so, then it makes sense to remove it, *unless* it might be used in the future.